Move clustering plots to a separate section

allyhawkins opened the pull request:
Stacked on #950 
Closes #948 

Here I'm moving the clustering UMAP and heatmap to a separate `Unsupervised clustering` section to be shown in the supplemental report. This section appears after the cell label comparisons (see #950) and before the diagnostic plots. I included a statement warning users to interpret the clusters with caution. 

At first, I was skeptical about moving the UMAP with clusters out of the main report since people love to look at colorful UMAPs, BUT I don't think it's entirely helpful where it currently is in the main report. We include it within the cell type section as a UMAP before the faceted UMAPs. I think _if_ we include the UMAP in the main report, it should be its own section outside of the cell type section. Because we don't really trust these clusters anyways, I don't love having a whole section in the main report on it, but feel free to convince me otherwise. 

So all of that led me back to just having the one section in the supplemental report. What do we think? 

Here's a copy of both reports: 
[reports.zip](https://github.com/user-attachments/files/21634074/reports.zip)

sjspielman commented:
> Because we don't really trust these clusters anyways, I don't love having a whole section in the main report on it, but feel free to convince me otherwise.

I agree with this and won't be trying to convince you otherwise (as the originating issue #948 suggests!), but let's give @jashapiro a chance to play devil's advocate here too?

sjspielman commented on templates/qc_report/celltypes_supplemental_report.rmd:
```diff
@@ -336,6 +336,61 @@ jaccard_consensus_matrices |>
+
+Clusters were calculated using the graph-based {metadata(processed_sce)$cluster_algorithm} algorithm with  {metadata(processed_sce)$cluster_weighting} weighting.
+
+**Caution**: Clusters are computed without performing any evaluation or optimization of clustering parameters.
```
either way, 💯

allyhawkins commented:
@sjspielman I updated this PR based on our discussion in pre-planning. The clusters UMAP is now in the dimensionality reduction and appears before the total genes detected and faceted HVGs. When we add tabs to the report, I imagine each of these will be its own tab. 

The supplemental report now has a clustering section with both the UMAP and the comparison heatmaps. 

When I was doing this I noticed that we had a function `plot_umap` that we were only using to create the clusters umap. I updated that to be `plot_clusters_umap` and included all the code to produce the UMAP from the processed SCE rather than the `umap_df`. This is because we don't make `umap_df` unless we have cell types, and it includes all the cell type information as well as clusters and UMAPs. Since this is now being created regardless of cell types, it made sense to just pull it from the `processed_sce` directly, like we do for the other UMAPs. 

Here's a new copy of the reports: 
[reports.zip](https://github.com/user-attachments/files/21722444/reports.zip)

allyhawkins commented:
@sjspielman I updated the text in the points you noted. I did not update the function though since I don't think it's worth the effort as mentioned in https://github.com/AlexsLemonade/scpca-nf/pull/951#discussion_r2270348799. This should be ready for another look.

sjspielman reviewed (approved):
LGTM!

allyhawkins commented:
Noting that I cherry-picked commits from #957 that @jashapiro used to fix the GHA since those changes are going into a different branch.
